pandas/core/window.py

```diff
@@ -341,6 +341,122 @@ def _center_window(self, result, window) -> np.ndarray:
                 result = np.copy(result[tuple(lead_indexer)])
         return result
 
+    def _wrap_roll(self, cfunc, check_minp, index, **kwargs) -> Callable:
+        """
+        Wrap rolling function.
+
+        Parameters
+        ----------
+        cfunc : callable
+            Cython function used to calculate rolling statistics
+        check_minp : callable
+            function to check minimum period parameter
+        index : ndarray
+            used for variable window
+        """
+
+        def func(arg, window, min_periods=None, closed=None):
+            minp = check_minp(min_periods, window)
+            return cfunc(arg, window, minp, index, closed, **kwargs)
+
+        return func
+
+    def _apply(
+        self, func, name=None, window=None, center=None, check_minp=None, **kwargs
+    ):
+        """
+        Rolling statistical measure using supplied function.
+
+        Designed to be used with passed-in Cython array-based functions.
+
+        Parameters
+        ----------
+        func : str/callable to apply
+        name : str, optional
+           name of this function
+        window : int/array, default to _get_window()
+        center : bool, default to self.center
+        check_minp : function, default to _use_window
+
```

Comment:
can you add what kwargs is here

Reply:
should we modify kwargs (or copy of it) to remove window function (passed to _get_window) kwargs first. window rolling functions (passed to _get_roll_func) don't take kwargs but when they do it will be a problem

Reply:
should we modify kwargs (or copy of it) to remove window function (passed to _get_window) kwargs first. window rolling functions (passed to _get_roll_func) don't take kwargs but when they do it will be a problem